core/app/models/spree/promotion.rb

```diff
@@ -112,13 +112,11 @@ def activate(order:, line_item: nil, user: nil, path: nil, promotion_code: nil)
       # If an action has been taken, report back to whatever activated this promotion.
       action_taken = results.include?(true)
 
-      if action_taken
-        # connect to the order
-        order_promotions.find_or_create_by!(
-          order_id: order.id,
-          promotion_code_id: promotion_code.try!(:id)
-        )
-      end
+      # connect to the order
+      order_promotions.find_or_create_by!(
+        order_id: order.id,
+        promotion_code_id: promotion_code.try!(:id)
+      )
 
       action_taken
```

Comment:
Is anything in solidus using the return value? (Guessing not). Another more or less backwards compatible option would be false for nothing happened (is that even still possible?), but :active and :inactive for attached in each of those states. 

Or, if the return value isn't meant to be relied upon as stable API, just return nil always. Or maybe true always to be more backwards compat to people who were using it anyway.